cluster/manager.go

```diff
@@ -738,77 +720,97 @@ func (c *ClusterManager) startListeners(db kvdb.Kvdb, exist *bool) (
 		db.Unlock(kvlock)
 	}()
 
-	initState, err := snapAndReadClusterInfo()
+	clusterInfo, err := readClusterInfo()
 	if err != nil {
 		dlog.Panicln(err)
 	}
-	defer func() {
-		if initState.Collector != nil {
-			initState.Collector.Stop()
-		}
-	}()
 
-	selfNodeEntry, ok := initState.ClusterInfo.NodeEntries[c.config.NodeId]
+	selfNodeEntry, ok := clusterInfo.NodeEntries[c.config.NodeId]
 	if ok && selfNodeEntry.Status == api.Status_STATUS_DECOMMISSION {
 		msg := fmt.Sprintf("Node is in decommision state, Node ID %s.",
 			c.selfNode.Id)
 		dlog.Errorln(msg)
-		return nil, nil, errors.New(msg)
+		return nil, true, nil, errors.New(msg)
 	}
-
-	// Cluster database max size... 0 if unlimited.
-	c.size = initState.ClusterInfo.Size
 	// Set the clusterID in db
-	initState.ClusterInfo.Id = c.config.ClusterId
+	clusterInfo.Id = c.config.ClusterId
+	self, nodeExists := c.initNode(&clusterInfo)
```

Comment:
initNode calls getCurrentState, which was fine until we added call in getCurrentState to get listener states. Listeners are not yet initialized - so I suggest removing getCurrentState from initNode.